### PR TITLE
feat: navigation bootstrap

### LINKED ISSUE #232: [FEAT]: Add nav/state-management on the hmi
**Description**

* Setup a state-management on the hmi to handle common information using Pinia
* Handle all this information within the URL
* Change `vue-router` to use [HTML5 mode](https://router.vuejs.org/guide/essentials/history-mode.html#html5-mode)

### PR DESCRIPTION
**Pull request by mwdchang**
# Description
This adds basic navigation plumbings following the discussions we have had

```
/ - home
/projects/:projectId - project
/projects/:projectId/model - model
/projects/:projectId/simulation - simulation
...
```

Note this is partially done because some pages are partially constructed or are placeholders with real routes, basically a cleanup round will be necessary to put everything together.

@YohannParis @echl @amostafa-uncharted 
Of note is that input and output are not synchronized because of the way the routes are named, so for example we have
```
/projects/:projectId
/docs/:docId
```
Which I have parsed as `view/viewId/subView/subViewId` in order to generalize.

But for the router to work you'd need to do `router.push({name: 'projects', params: { projectId: xyz })` and `router.push({ name: 'docs' })` respectively, so there is some 

Not sure if this is confusing or now, of if the names view/viewId are good .... your thoughts and comments are welcome!

# Testing
1. Click the mock project, this will mock a project called `123`
2. Click the simulation/model to toggle between views without losing the project context (the top-2 buttons in the side bar)

Resolves #232

### CODE REVIEW

**Comment by YohannParis** [comment]
If you imported my branch into yours, it might be best to direct this PR to `sidebar-improvments`, so which ever is done first, it's easier to review, and both end up in `main`

File: packages/client/hmi-client/src/components/Header.vue [hunk 1]
```diff
@@ -1,24 +1,36 @@
...
 
 const emit = defineEmits(['show-data-explorer']);
-
 const router = useRouter();
+const { isCurrentRouteHome, routerState } = useCurrentRouter();
+const isHome = computed(() => isCurrentRouteHome.value);
 
-const goToHomepage = () => router.push('/');
+const goToHomepage = () => {
+	console.log('goto home');
```
Comment (amostafa-uncharted) [inline packages/client/hmi-client/src/components/Header.vue:14]: remove the console statment

File: packages/client/hmi-client/src/components/Sidebar.vue [hunk 2]
```diff
@@ -24,75 +37,71 @@ const enum Mode {
...
-
-function moveSidebar() {
-	localStorage.setItem('isSidebarPositionRight', (!isSidebarPositionRight.value).toString());
-	isSidebarPositionRight.value = localStorage.getItem('isSidebarPositionRight') === 'true';
+/**
+ * Open a View
+ * @param {string} view - The view to be open.
+ * @param {boolean} [openViewSidePanel=true] - Should the side-panel be open when opening the view.
+ */
+function openView(view: string, openViewSidePanel: boolean = true): void {
```
Comment (amostafa-uncharted) [inline packages/client/hmi-client/src/components/Sidebar.vue:47]: Can we convert "view" to be an enum instead of raw string?

File: packages/client/hmi-client/src/components/Sidebar.vue [hunk 3]
```diff
@@ -24,75 +37,71 @@ const enum Mode {
...
+	selectedView.value = view;
+	if (isSidePanelOpen.value) {
+		if (!openViewSidePanel) closeSidePanel();
+	} else if (openViewSidePanel) openSidePanel();
+
+	if (view === View.SimulationPlan) {
+		router.push({ name: 'simulation' });
```
Comment (amostafa-uncharted) [inline packages/client/hmi-client/src/components/Sidebar.vue:54]: can we just use the value of "name" to be the equivalent enum value instead of hard-coding the "simulation" string

File: packages/client/hmi-client/src/components/sidebar-panel/documents-sidebar-panel.vue [hunk 4]
```diff
@@ -1,24 +1,18 @@
...
-				>
-					<span>{{ formatTitle(doc) }}</span>
-					<span class="doc-delete-btn" @click.stop="removeDocument(doc)">
-						<IconClose32 />
-					</span>
-				</div>
-			</div>
-		</section>
-	</main>
+	<div class="document-list-container">
```
Comment (amostafa-uncharted) [inline packages/client/hmi-client/src/components/sidebar-panel/documents-sidebar-panel.vue:2]: Why did you remove the side panel header text?

**Comment by amostafa-uncharted** [comment]
I left a few minor comments, but it looks good overall. You may address them, resolve and merge

File: packages/client/hmi-client/src/router/index.ts [hunk 5]
```diff
@@ -41,7 +46,17 @@ const router = createRouter({
...
+		isCurrentRouteHome: computed(() => router.currentRoute.value.path === RoutePath.Home),
+		routerState: computed(() => {
+			const path = router.currentRoute.value.path;
+
+			// This maps into the navigation hierarchy :view/:viewId/:subView/:subViewId, eg
+			// - /projects/123/simulations/232
+			// - /projects/222/models/456
+			// - /docs/d123
+			const [, view, viewId, subView, subViewId] = path.split('/');
+			return { view, viewId, subView, subViewId };
```
Comment (YohannParis) [inline packages/client/hmi-client/src/router/index.ts:58]: `vue-router` doesn't provide a better way to do this than `path.split('/')`?
Can we use RegEx to have at least better control on routing the user. I feel like returning an object seems so overkill.

File: packages/client/hmi-client/src/components/sidebar-panel/documents-sidebar-panel.vue (hunk 4 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/components/sidebar-panel/documents-sidebar-panel.vue]: I did not - this is build on top of @YohannParis 's sidebar changes.

File: packages/client/hmi-client/src/router/index.ts (hunk 5 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/router/index.ts:58]: I think if you have canonical path-params you can sort of use vue-router to just tell you what the current ids are, but since there are some overlap it can be ambiguous.

I don't know what you mean by using regular expressions. Can you elaborate?

Comment (YohannParis) [reply packages/client/hmi-client/src/router/index.ts:58]: like this: https://v3.router.vuejs.org/guide/essentials/dynamic-matching.html#dynamic-route-matching

File: packages/client/hmi-client/src/components/Sidebar.vue (hunk 3 above)
Comment (amostafa-uncharted) [reply packages/client/hmi-client/src/components/Sidebar.vue:56]: No worries. Not a priority anyway!

File: packages/client/hmi-client/src/router/index.ts (hunk 5 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/router/index.ts:72]: But I am using the matching patterns 😕 
```
	ModelView = '/projects/:projectId/model',
	SimulationView = '/projects/:projectId/simulation',
```

**Review by YohannParis — DISMISSED** [review]

File: packages/client/hmi-client/src/router/index.ts (hunk 5 above)
Comment (YohannParis) [reply packages/client/hmi-client/src/router/index.ts]: yes, but this comment is outdated now

File: packages/client/hmi-client/src/components/Header.vue [hunk 6]
```diff
@@ -20,9 +22,13 @@ const projectName = 'Name of the project that can be long for clarity and precis
 	<header>
 		<img v-if="isHome" src="@assets/images/logo.png" height="32" width="128" alt="logo" />
 		<img v-else src="@assets/images/icon.png" height="32" width="32" alt="TERArium icon" />
-		<p>
-			<a @onclick="goToHomepage">Projects</a>
-			<span>{{ projectName }}</span>
+		<p v-if="!isHome">
+			<a @click="goToHomepage">Projects</a>
+			<span> {{ projectName }}</span>
+			<!-- Debug -->
```
Comment (echl) [inline packages/client/hmi-client/src/components/Header.vue:28]: Do we need this debug stuff here still?

File: packages/client/hmi-client/src/views/Home.vue [hunk 7]
```diff
@@ -30,7 +37,7 @@ const mockProjects: string[] = ['Mocked Project'];
 					<li v-if="key === Categories.Recents">
 						<NewProjectCard />
 					</li>
-					<li v-for="(project, index) in mockProjects" :key="index">
+					<li v-for="(project, index) in mockProjects" :key="index" @click="gotoModel()">
```
Comment (echl) [inline packages/client/hmi-client/src/views/Home.vue:40]: Shouldn't clicking on the project card open project page? In that case you could leave this change out as it would be taken care of in https://github.com/DARPA-ASKEM/TERArium/pull/235

File: packages/client/hmi-client/src/components/Header.vue (hunk 6 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/components/Header.vue:28]: Yes - we will remove it prior to the final demo build. Right now it is useful for debugging.

File: packages/client/hmi-client/src/views/Home.vue (hunk 7 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/views/Home.vue]: removed

**Review by echl — APPROVED** [review]

**Review by YohannParis — APPROVED** [review]
